flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -3203,6 +3212,51 @@ class ArrayExprLowering {
     return {indices, loops[0]};
   }
 
+  llvm::SmallVector<mlir::Value> genImplicitLoopBounds(
+      const Fortran::evaluate::Expr<Fortran::evaluate::SomeType> &e) {
+    struct Filter : public Fortran::evaluate::AnyTraverse<
```

Comment:
What is guaranteeing that the `ArrayRef` in `e` that gets visited is the one that matters here ?

Reply:
You're right. This is a fake "analysis" here. I think I'll have to write one from scratch to get what I really need.